impress-org/givewp
# Refactor: normalize currency component

## Description

This Pull Request introduces enhanced consistency in value formatting for the `onValueChange` method in the `CurrencyControl` component by leveraging an internal state. It ensures that all values are normalized before being passed to the method. Furthermore, the `CurrencyControl` component is now globally accessible through `window.givewp.form.components.CurrencyControl`, making it easier for add-ons to utilize this controlled component.

Prior to these changes, input values were stored in the form builder just as they were entered by the user, localized according to each user's locale. This led to unpredictable and inconsistent data formats. For example, one user might edit the Fee Recovery's percentage fee field using a comma, while another could edit only the base amount field using a dot.

To resolve this issue, an internal state has been added to the component. When the input value changes, this internal state will be manually updated with the raw, unformatted value. Simultaneously, the block attribute will be set with the normalized value. This ensures consistent data formatting across different locales and editing stages.

By adopting this approach, we're standardizing the data format, eliminating any inconsistencies, and simplifying the process for future edits and data manipulations.

## Affects

Fee Recovery settings and Donation Levels inputs

## Testing Instructions

- Create a form builder using different locales in your browser (Inspector Tools > Sensors)
- Check that saved data is stored normalized as number

## Pre-review Checklist
